src/components/dialogs/network-modifications/generator/generator-dialog.type.ts

```diff
@@ -0,0 +1,251 @@
+    FREQUENCY_REGULATION,
+    ID,
+    MARGINAL_COST,
+    MAX_Q,
+    MAXIMUM_ACTIVE_POWER,
+    MAXIMUM_REACTIVE_POWER,
+    MIN_Q,
+    MINIMUM_ACTIVE_POWER,
+    MINIMUM_REACTIVE_POWER,
+    NAME,
+    NOMINAL_VOLTAGE,
+    P,
+    PLANNED_ACTIVE_POWER_SET_POINT,
+    PLANNED_OUTAGE_RATE,
+    Q_PERCENT,
+    RATED_NOMINAL_POWER,
+    REACTIVE_CAPABILITY_CURVE_CHOICE,
+    REACTIVE_CAPABILITY_CURVE_TABLE,
+    REACTIVE_LIMITS,
+    REACTIVE_POWER_SET_POINT,
+    SUBSTATION_ID,
+    TOPOLOGY_KIND,
+    TRANSFORMER_REACTANCE,
+    TRANSIENT_REACTANCE,
+    TYPE,
+    VOLTAGE_LEVEL,
+    VOLTAGE_REGULATION,
+    VOLTAGE_REGULATION_TYPE,
+    VOLTAGE_SET_POINT,
+} from '../../../utils/field-constants';
+import { EquipmentType } from '@gridsuite/commons-ui';
+import { Property } from '../common/properties/property-utils';
+import { ConnectablePositionFormInfos } from '../../connectivity/connectivity.type';
+
+export type ValueWrapper<T> = { value: T };
+
+export type ReactiveCapabilityCurveTable = {
+    [P]: number | null;
+    [MAX_Q]: number | null;
+    [MIN_Q]: number | null;
+};
+
+export type GeneratorModificationSchemaForm = {
+    [EQUIPMENT_ID]: string;
+    [EQUIPMENT_NAME]?: string;
+    [ENERGY_SOURCE]?: string | null;
+    [MAXIMUM_ACTIVE_POWER]?: number | null;
+    [MINIMUM_ACTIVE_POWER]?: number | null;
+    [RATED_NOMINAL_POWER]?: number;
+    [TRANSFORMER_REACTANCE]?: number;
+    [TRANSIENT_REACTANCE]?: number | null;
+    [PLANNED_ACTIVE_POWER_SET_POINT]?: number | null;
+    [MARGINAL_COST]?: number | null;
+    [PLANNED_OUTAGE_RATE]?: number | null;
+    [FORCED_OUTAGE_RATE]?: number | null;
+
+    [CONNECTIVITY]: {
+        [VOLTAGE_LEVEL]: { [ID]?: string };
+        [BUS_OR_BUSBAR_SECTION]: { [ID]?: string; [NAME]?: string };
+        [CONNECTION_DIRECTION]?: string;
+        [CONNECTION_NAME]?: string;
+        [CONNECTION_POSITION]?: number;
+        [CONNECTED]?: boolean;
+    };
+
+    [VOLTAGE_REGULATION]?: boolean | null;
+    [ACTIVE_POWER_SET_POINT]?: number;
+    [REACTIVE_POWER_SET_POINT]?: number | null;
+    [VOLTAGE_REGULATION_TYPE]?: string | null;
+    [VOLTAGE_SET_POINT]?: number | null;
+    [Q_PERCENT]?: number | null;
+
+    [VOLTAGE_LEVEL]?: {
+        [ID]?: string;
+        [NAME]?: string;
+        [SUBSTATION_ID]?: string;
+        [NOMINAL_VOLTAGE]?: string;
+        [TOPOLOGY_KIND]?: string | null;
+    };
+
+    [EQUIPMENT]?: {
+        [ID]?: string;
+        [NAME]?: string | null;
+        [TYPE]?: string;
+    };
+    [FREQUENCY_REGULATION]?: boolean | null;
+    [DROOP]?: number | null;
+    [REACTIVE_LIMITS]: {
+        [MINIMUM_REACTIVE_POWER]?: number | null;
+        [MAXIMUM_REACTIVE_POWER]?: number | null;
+        [REACTIVE_CAPABILITY_CURVE_CHOICE]: string | null;
+        [REACTIVE_CAPABILITY_CURVE_TABLE]?: ReactiveCapabilityCurveTable[];
+    };
+    // Properties
+    [ADDITIONAL_PROPERTIES]?: Property[];
+};
+
+export interface GeneratorCreationInfos {
+    uuid: string;
```

Comment:
```suggestion
    uuid: UUID;
```
for uuid, using type UUID is suggested.. if typing problem occurs between string and UUID => use 'as'

Reply:
done